packages/docusaurus/src/client/client-lifecycles-dispatcher.ts

```diff
@@ -20,9 +20,11 @@ function dispatchLifecycleAction(
   ...args: any[]
 ) {
   clientModules.forEach((clientModule) => {
-    const mod = clientModule.__esModule ? clientModule.default : clientModule;
-    if (mod && mod[lifecycleAction]) {
-      mod[lifecycleAction](...args);
+    const lifecycleFunction =
```

Comment:
I'm not familiar with this code, but surely removing the `clientModule.__esModule` won't affect it in any way? It was kind of related to Yarn 2 support (not sure)

Reply:
This is a behavior change that shouldn't break anything, but allows you to write client lifecycle methods as named exports.

Before we had to use this syntax to add the lifecycle to default expor

```
export default {
    onRouteUpdate({location}) {
      console.log('onRouteUpdate', {location});
    },
};
```

Now we can also write the following:

```
export function onRouteUpdate({location}: {location: Location}) {
  console.log('onRouteUpdate', {location});
}
```

Note it's not a documented feature (yet) and kind of broken, so I plan to improve this as part of  https://github.com/facebook/docusaurus/issues/3399

Reply:
BTW both versions still work, and it's the default export that is picked in priority if both are used, so it should be retrocompatible in all cases even if it's not very likely that such code exist in userland

```js
export function onRouteUpdate({location}: {location: Location}) {
  console.log('onRouteUpdate', {location});
}

export default {
  onRouteUpdate: function onRouteUpdate({location}: {location: Location}) {
    console.log('default onRouteUpdate', {location});
  },
};
```

Reply:
Thanks for the explanation, so this is safe for merge.